src/models/user.ts

```diff
@@ -160,6 +161,25 @@ export class User extends TypedEventEmitter<UserEvent, UserEventHandlerMap> {
         this.updateModifiedTime();
     }
 
+    /**
+     * Construct a new User whose events will also emit on MatrixClient.
+     * A User must have an ID and can optionally have extra information associated with it.
+     * @param userId - Required. The ID of this user.
+     * @param client - An instance of MatrixClient object
+     * @returns User object with reEmitter setup on client
+     */
+    public static withReEmitterOnClient(userId: string, client: MatrixClient): User {
```

Comment:
I would also suggest making the `User` constructor deprecated and commenting there that folks should use `User.createUser` instead to have all the appropriate events set up

<img width="455" alt="Screenshot 2023-10-23 at 07 10 49" src="https://github.com/matrix-org/matrix-js-sdk/assets/769871/a43fe075-0c3e-4698-9ec4-4d71d568d658">

```
     * @deprecated use `User.withReEmitterOnClient`
```